server/odc-service/src/main/java/com/oceanbase/odc/service/connection/logicaldatabase/parser/LogicalTableExpressionVisitor.java

```diff
@@ -0,0 +1,172 @@
+        }
+        expressions.setExpressions(expressionList);
+        return expressions;
+    }
+
+    @Override
+    public BaseStatement visitLogicalTableExpression(LogicalTableExpressionContext ctx) {
+        PreConditions.notNull(ctx.schemaExpression(), "schemaExpression");
+        PreConditions.notNull(ctx.tableExpression(), "tableExpression");
+        LogicalTableExpression expression = new LogicalTableExpression(ctx);
+        expression.setSchemaExpression((SchemaExpression) visitSchemaExpression(ctx.schemaExpression()));
+        expression.setTableExpression((TableExpression) visitTableExpression(ctx.tableExpression()));
+        return expression;
+    }
+
+    @Override
+    public BaseStatement visitSchemaExpression(SchemaExpressionContext ctx) {
+        PreConditions.notEmpty(ctx.IDENTIFIER(), "schemaExpression.IDENTIFIER");
+        SchemaExpression expression = new SchemaExpression(ctx);
+        if (ctx.schemaSliceRangeWithBracket() != null) {
+            List<BaseRangeExpression> ranges = new ArrayList<>();
+            for (SchemaSliceRangeWithBracketContext sliceRange : ctx.schemaSliceRangeWithBracket()) {
+                ranges.add((BaseRangeExpression) visitSchemaSliceRangeWithBracket(sliceRange));
+            }
+            expression.setSliceRanges(ranges);
+        }
+        return expression;
+    }
+
+    @Override
+    public BaseStatement visitTableExpression(TableExpressionContext ctx) {
+        PreConditions.notEmpty(ctx.IDENTIFIER(), "tableExpression.IDENTIFIER");
+        TableExpression expression = new TableExpression(ctx);
+        if (ctx.tableSliceRangeWithBracket() != null) {
+            List<BaseRangeExpression> ranges = new ArrayList<>();
+            for (TableSliceRangeWithBracketContext sliceRange : ctx.tableSliceRangeWithBracket()) {
+                if (sliceRange.sliceRangeWithDoubleBracket() != null) {
+                    expression.setRepeat(true);
+                }
+                ranges.add((BaseRangeExpression) visitTableSliceRangeWithBracket(sliceRange));
+            }
+            expression.setSliceRanges(ranges);
+        }
+        return expression;
+    }
+
+    @Override
+    public BaseStatement visitSchemaSliceRangeWithBracket(SchemaSliceRangeWithBracketContext ctx) {
+        if (ctx.sliceRangeWithSingeBracket() != null) {
+            return visitSliceRangeWithSingeBracket(ctx.sliceRangeWithSingeBracket());
+        }
+        return null;
+    }
+
+    @Override
+    public BaseStatement visitTableSliceRangeWithBracket(TableSliceRangeWithBracketContext ctx) {
+        if (ctx.sliceRangeWithSingeBracket() != null) {
+            return visitSliceRangeWithSingeBracket(ctx.sliceRangeWithSingeBracket());
+        } else if (ctx.sliceRangeWithDoubleBracket() != null) {
+            return visitSliceRangeWithDoubleBracket(ctx.sliceRangeWithDoubleBracket());
+        } else {
+            return null;
```

Comment:
this branch will never be run

Reply:
Anyway, this method need to return a value.